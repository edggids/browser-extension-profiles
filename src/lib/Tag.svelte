<script lang="ts">
  import {createEventDispatcher} from "svelte";

  let className = '';
  export { className as class };

  export let active = false;

  export let color = '#1d1e1e';

  const dispatch = createEventDispatcher();

  function onClick() { dispatch('click'); }
</script>

<span on:click={onClick} class:active class={ `tag transition-border-color transition-color rounded-xl p-2.75 leading-none pr-4 inline-flex items-center text-sm bg-ground ${className}` } style:--color={color}>
  <span class="indicator mr-2.75"></span><slot></slot>
</span>
