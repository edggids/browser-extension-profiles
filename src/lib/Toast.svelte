<script lang="ts">
    import type {Toast} from "../store/toasts";
    import {createEventDispatcher} from "svelte";

    export let toast: Toast;

    let className = '';
    export { className as class };

    const colorClassMap: Record<Toast['state'], { bg: string, text: string }> = {
        info: { bg: 'bg-safe-light', text: 'text-safe-darker' },
        error: { bg: 'bg-danger-light', text: 'text-danger-darker' },
        warning: { bg: 'bg-danger-light', text: 'text-danger-darker' },
    };

    $: role = toast.state === 'error' ? 'alert' : 'status';
    $: colorClass = colorClassMap[toast.state] ?? colorClassMap.info;

    const dispatch = createEventDispatcher();

    const onClick = (toast: Toast) => dispatch('close', { toast });
</script>

<div role={role} class={`${colorClass.bg} ${colorClass.text} pl-4 pt-3 pb-3 text-sm flex items-center rounded-xs ${className}`} >
    <span class="flex-1 mr-3.5 pb-1px">{toast.message}</span>
    <button data-testid="closeButton" type="button" class={`${colorClass.text} pr-4 pl-4`} on:click={e => onClick(toast)}>x</button>
</div>
