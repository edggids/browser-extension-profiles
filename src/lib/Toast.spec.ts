import {cleanup, render, fireEvent} from '@testing-library/svelte';
import {describe, it, expect, afterEach, vi} from 'vitest';
import Toast from "./Toast.svelte";
import Tag from "./Tag.svelte";

describe('Toast component', () => {
  afterEach(() => {
    cleanup();
  });

  it('mounts', () => {
    const { container } = render(Toast, { props: { toast: { message: 'Test Message', state: 'info' } } });

    expect(container).toMatchSnapshot();
  });

  it('renders with provided message', () => {
    const { getByText } = render(Toast, { props: { toast: { message: 'Test Message', state: 'info' } } });
    expect(getByText('Test Message')).not.toBeNull();
  });

  it('renders with correct role for info state', () => {
    const { getByRole } = render(Toast, { props: { toast: { message: 'Test Message', state: 'info' } } });
    expect(getByRole('status')).not.toBeNull();
  });

  it('renders with correct role for error state', () => {
    const { getByRole } = render(Toast, { props: { toast: { message: 'Test Message', state: 'error' } } });
    expect(getByRole('alert')).not.toBeNull();
  });

  it('emits close event when close button is clicked', async () => {
    const { getByTestId, component } = render(Toast, { props: { toast: { message: 'Test Message', state: 'info' } } });

    const closeListener = vi.fn();
    component.$on('close', closeListener);

    await fireEvent.click(getByTestId('closeButton'));
    expect(closeListener).toHaveBeenCalled();
  });
});
