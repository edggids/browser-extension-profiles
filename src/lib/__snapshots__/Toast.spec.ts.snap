// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Toast component > mounts 1`] = `
<body>
  <div>
    <div
      class="bg-safe-light text-safe-darker pl-4 pt-3 pb-3 text-sm flex items-center rounded-xs "
      role="status"
    >
      <span
        class="flex-1 mr-3.5 pb-1px"
      >
        Test Message
      </span>
       
      <button
        class="text-safe-darker pr-4 pl-4"
        data-testid="closeButton"
        type="button"
      >
        x
      </button>
    </div>
  </div>
</body>
`;
